### PR TITLE
Add support to update environment variables

### PR DESCRIPTION
**Pull request by pragnagopa**
Adds support to reload environment variables

### CODE REVIEW

File: src/main/java/com/microsoft/azure/functions/worker/handler/FunctionEnvironmentReloadRequestHandler.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    return String.format("FunctionEnvironmentReloadRequest completed");
+  }
+
+  FunctionMethodDescriptor createFunctionDescriptor(String functionId,
+      RpcFunctionMetadata metadata) {
+    return new FunctionMethodDescriptor(functionId, metadata.getName(), metadata.getEntryPoint(),
+        metadata.getScriptFile());
+  }
+  
+  public static void setEnv(Map<String, String> newSettings) throws Exception {
```
Comment (pragnagopa) [inline src/main/java/com/microsoft/azure/functions/worker/handler/FunctionEnvironmentReloadRequestHandler.java]: @fabiocav - Java does not have a built in helper to set the environment variables. Following code snippet is taken from [SO post](https://stackoverflow.com/questions/318239/how-do-i-set-environment-variables-from-java)

File: src/main/azure-functions-language-worker-protobuf/src/proto/FunctionRpc.proto [hunk 2]
```diff
@@ -5,6 +5,7 @@ option java_multiple_files = true;
 option java_package = "com.microsoft.azure.functions.rpc.messages";
 option java_outer_classname = "FunctionProto";
 option csharp_namespace = "Microsoft.Azure.WebJobs.Script.Grpc.Messages";
+option go_package ="github.com/Azure/azure-functions-go-worker/internal/rpc";
```
Comment (fabiocav) [inline src/main/azure-functions-language-worker-protobuf/src/proto/FunctionRpc.proto]: Can you help me understand what this is for?

File: src/main/java/com/microsoft/azure/functions/worker/Application.java [hunk 3]
```diff
@@ -122,7 +123,7 @@ public static void main(String[] args) {
             try (JavaWorkerClient client = new JavaWorkerClient(app)) {
                 client.listen(app.getWorkerId(), app.getRequestId()).get();
             } catch (Exception ex) {
-                WorkerLogManager.getSystemLogger().log(Level.SEVERE, "Unexpected Exception causes system to exit", ex);
+                WorkerLogManager.getSystemLogger().log(Level.SEVERE, ExceptionUtils.getRootCauseMessage(ex), ex);
```
Comment (fabiocav) [inline src/main/java/com/microsoft/azure/functions/worker/Application.java]: It this an unrelated change?

File: src/main/java/com/microsoft/azure/functions/worker/handler/FunctionEnvironmentReloadRequestHandler.java (hunk 1 above)
Comment (fabiocav) [reply src/main/java/com/microsoft/azure/functions/worker/handler/FunctionEnvironmentReloadRequestHandler.java]: Wow... this looks pretty fragile. Should we have a note here?

File: src/main/azure-functions-language-worker-protobuf/src/proto/FunctionRpc.proto (hunk 2 above)
Comment (pragnagopa) [reply src/main/azure-functions-language-worker-protobuf/src/proto/FunctionRpc.proto]: as of now go support is in private preview.

File: src/main/java/com/microsoft/azure/functions/worker/Application.java (hunk 3 above)
Comment (pragnagopa) [reply src/main/java/com/microsoft/azure/functions/worker/Application.java]: Yes. Improving error logging to help us get the exact exception message
